test/cron_test.rb

```diff
@@ -0,0 +1,116 @@
+require_relative 'test_helper'
+require 'rack'
+require 'json'
+require 'open-uri'
+
+SingleCov.not_covered!
+
+describe 'CRON' do
+  class FakeApiServer
+    TEST_ENDPOINT = "/__ping__".freeze
+
+    def self.open(port, replies)
+      server = new(port, replies)
+      server.boot
+      yield server
+    ensure
+      server.shutdown
+    end
+
+    def initialize(port, replies)
+      @port = port
+      @replies = replies.merge(TEST_ENDPOINT => 'PONG')
+    end
+
+    def boot
+      Thread.new do
+        Rack::Handler::WEBrick.run(
+          self,
+          Port: @port,
+          Logger: WEBrick::Log.new("/dev/null"),
+          AccessLog: []
+        ) { |s| @server = s }
+      end
+    end
+
+    def wait
+      loop do
+        begin
+          break if URI.parse("http://localhost:#{@port}#{TEST_ENDPOINT}").read
+        rescue
+          nil
+        end
+      end
+    end
+
+    def call(env)
+      path = env.fetch("PATH_INFO")
+      unless reply = @replies[path]
+        puts "ERROR: Missing reply for path #{path}" # kubeclient does not show current url when failing
+        raise
+      end
+      [200, {}, [reply.to_json]]
+    end
+
+    def shutdown
+      @server.shutdown if @server
+    end
+  end
+
+  def with_env(env)
+    old = env.keys.map { |k| [k, ENV[k.to_s]] }
+    env.each { |k, v| ENV[k.to_s] = v }
+    yield
+  ensure
+    old.each { |k, v| ENV[k.to_s] = v }
+  end
+
+  def sh(command)
+    result = `#{command} 2>&1`
+    raise "FAILED #{result}" unless $?.success?
+    result
+  end
+
+  let(:vault_replies) do
+    {
+      '/auth/token/create' =>
+        {
+          "auth": {
+            "client_token": "ABCD",
+            "policies": ["web", "stage"],
+            "metadata": {"user": "armon"},
+            "lease_duration": 3600,
+            "renewable": true,
+          }
+        },
+      '/auth' => {}
+
+    }
+  end
+  let(:kuber_replies) do
+    {
+      '/api/v1/namespaces/default/secrets/vaultauth' => {},
+    }
+  end
+
+  around do |test|
+    WebMock.disable!
+    FakeApiServer.open(8200, vault_replies) do |vault|
+      FakeApiServer.open(8443, kuber_replies) do |kube|
+        vault.wait
+        kube.wait
+        with_env(
+          CRON_TEST: 'true',
+          BUNDLER_ROOT: Bundler.root.to_s,
+          &test
+        )
+        test.call
+      end
+    end
+  end
+
+  it "can run the cron job" do
+    br = Bundler.root.to_s
+    sh "#{br}/bin/create_k8s_token.sh -v http://localhost:8200 -t faketoken -k #{br}/test/fixutes/kube_config"
```

Comment:
added tests for invalid options/server not listening. successful run implies it all worked.